pkg/common/cns-lib/volume/manager.go

```diff
@@ -973,9 +973,22 @@ func (m *defaultManager) DetachVolume(ctx context.Context, vm *cnsvsphere.Virtua
 		volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
 		if volumeOperationRes.Fault != nil {
 			faultType = ExtractFaultTypeFromVolumeResponseResult(ctx, volumeOperationRes)
+
+			fault, isManagedObjectNotFoundFault := volumeOperationRes.Fault.Fault.(*vim25types.ManagedObjectNotFound)
```

Comment:
can we add a nil check for volumeOperationRes.Fault and volumeOperationRes.Fault.Fault

Reply:
nil check for volumeOperationRes.Fault is already available at line 974, added a nil check for volumeOperationRes.Fault.Fault